openmetadata-service/src/main/java/org/openmetadata/service/apps/scheduler/AppScheduler.java

```diff
@@ -177,7 +176,7 @@ public void deleteScheduledApplication(App app) throws SchedulerException {
 
   private JobDetail jobBuilder(App app, String jobIdentity) throws ClassNotFoundException {
     JobDataMap dataMap = new JobDataMap();
-    dataMap.put(APP_INFO_KEY, JsonUtils.pojoToJson(app));
+    dataMap.put(APP_ID_KEY, app.getId());
```

Comment:
@sushi30 we should work with AppName rather than appId I think

Reply:
@mohityadav766 by using appName we introduce 2 isues:
1. Risk of shading the appName being shaded by another app (for external applications).
2. We remove the capability to deploy multiple instances of the same app. (example: multiple re-indexing jobs for different languages)

Reply:
@sushi30 
we need to anyway have only a single application instance anyway, also user will never be able to register apps with same name

Reply:
@mohityadav766
ID is the way we link entities throughout the application. Using the `name` for the application will break this convention. I am yet to be persuaded regarding the reason.

1. An app id should not change for the lifecycle of the installation.
2. While we currently only support one instance of the applications, I think we should leave the door open for enabling more complex usages.
3. An app name can change in the future (not supported at the moment, but might be...). An ID is supposed to stay constant.